pkg/component/component.go

```diff
@@ -1612,6 +1618,22 @@ func setLinksServiceNames(client *occlient.Client, linkedSecrets []SecretMount)
 		}
 	}
 
+	if !ok {
+		secrets, err := client.GetKubeClient().ListSecrets(selector)
+		if err != nil {
+			return err
+		}
+		for _, secret := range secrets {
+			serviceName, serviceOK := secret.Labels[service.ServiceLabel]
+			_, linkOK := secret.Labels[service.LinkLabel]
+			if serviceOK && linkOK {
+				if _, ok := serviceBindings[secret.Name]; !ok {
+					serviceBindings[secret.Name] = serviceName
```

Comment:
I can see a difference between the service name found in the devfile (`Redis/myredis`) and the one detected here (`myredis`):

```
$ odo describe 
Component Name: api
Type: nodejs
Environment Variables:
 · PROJECTS_ROOT=/project
 · PROJECT_SOURCE=/project
 · DEBUG_PORT=5858
URLs:
 · http://http-3000-app-prj1.apps-crc.testing exposed via 3000
Linked Services:
 · myredis           <- detected from secret
   Environment Variables:
    · REDIS_CLUSTERIP
 · Redis/myredis     <- found in devfile.yaml
```

One solution would be to store some info about the service linked into an annotation of the Secret? @kadel